src/ODE.jl

```diff
@@ -251,6 +251,17 @@ function ode23s(F, y0, tspan; reltol = 1.0e-5, abstol = 1.0e-8,
 
     # initialization
     t = tspan[1]
+    
+    # Component-wise Tolerance check similar to MATLAB ode23s
+    # Support for component-wise absolute tolerance only.
+    # Relative tolerance should be a scalar.
+    @assert length(reltol) == 1 "Relative tolerance must be a scalar"
```

Comment:
Sure. But in the specific case handled by this line, since length 1 is hard-coded anyway, I think it would be much clearer.